feat: Prism Central and NKE install

cprivitere opened the pull request:
Signed-off-by: Chris Privitere <23177737+cprivitere@users.noreply.github.com>

cprivitere committed:
feat: draft of Prism Central and NKE install

Signed-off-by: Chris Privitere <23177737+cprivitere@users.noreply.github.com>

ctreatma commented on docs/parts/part5.md:
```diff
@@ -0,0 +1,83 @@
+<!-- See https://squidfunk.github.io/mkdocs-material/reference/ -->
+
+# Part 5: Deploy Prism Central
```
Could you add these pages in mkdocs.yaml so they come up with `mkdocs serve`?

ctreatma commented on docs/parts/part5.md:
```diff
@@ -0,0 +1,83 @@
+
+### 4. Reconnect your SSH tunnel to Prism Central
+
+After prism installs you'll need to reconnect with an updated ssh forward command that forwards to the Prism VM:
```
It could be helpful to either explain here why they need to update this or add a discussion question about it.  I think I lean towards the discussion question

ctreatma commented on docs/parts/part5.md:
````diff
@@ -0,0 +1,83 @@
+After prism installs you'll need to reconnect with an updated ssh forward command that forwards to the Prism VM:
+
+```sh
+ssh -L 9440:192.168.101.47:9440 -i $(terraform output -raw ssh_private_key) root@(bastion_public_ip)
````
```suggestion
ssh -L 9440:192.168.101.47:9440 -i $(terraform output -raw ssh_private_key) root@$(terraform output -raw bastion_public_ip)
```

Haven't tested this yet,

cprivitere replied:
Yeah it needs to be updated with the reserved IP

cprivitere committed:
docs: add parts 5 and 6 to mkdocs file.

Signed-off-by: Chris Privitere <23177737+cprivitere@users.noreply.github.com>

cprivitere committed:
docs: add discussion question about new ssh command

Signed-off-by: Chris Privitere <23177737+cprivitere@users.noreply.github.com>

cprivitere committed:
docs: major updates to text and add screenshots

Signed-off-by: Chris Privitere <23177737+cprivitere@users.noreply.github.com>

cprivitere committed:
docs: update to nke from karbon, finish kubernetes docs pictures

Signed-off-by: Chris Privitere <23177737+cprivitere@users.noreply.github.com>

cprivitere committed:
chore: run prettier on docs

Signed-off-by: Chris Privitere <23177737+cprivitere@users.noreply.github.com>

ocobles commented on docs/parts/prism_central.md:
```diff
@@ -0,0 +1,108 @@
+
+# Part 5: Deploy Prism Central
+
+Many of the more powerful capabilities of Nutanix are only unlocked after configuring Prism Central. In this part of the workshop, we will deploy Prism Central on our Nutanix cluster.
```
Unless there are some restrictions with us adding external links, I would refer to the Prism Central documentation. I'm not sure if there's a better link I just found that one

```suggestion
Many of the more powerful capabilities of Nutanix are only unlocked after configuring [Prism Central](https://portal.nutanix.com/page/documents/details?targetId=Prism-Central-Guide-vpc_2023_4:mul-pc-overview-c.html). In this part of the workshop, we will deploy Prism Central on our Nutanix cluster.
```

ocobles commented:
nit: keeping the files in the `parts` folder with numbering makes the workshop maintenance easier

cprivitere committed:
Update docs/parts/prism_central.md

Co-authored-by: Óscar Cobles <68897552+ocobles@users.noreply.github.com>

ctreatma commented on docs/parts/prism_central.md:
```diff
@@ -0,0 +1,108 @@
+
+![VM Network](../images/vm-network.png)
+
+Now enter `255.255.252.0` for the Subnet Mask and `192.168.100.2` for the Gateway, and `192.168.103.252` for the VM IP and click `Next`.
```
For me there was an extra button click here.  `Next` took me to a summary page, and clicking `Deploy` on that page started the deployment

ctreatma commented on docs/parts/prism_central.md:
```diff
@@ -0,0 +1,108 @@
+
+![Prism Central Registration](../images/prism-central-registration.png)
+
+Click `Next` on the `Connect Info` page and then put in the Prism Central IP address `192.168.103.252`, port of `9440`, and the username and password you used to log in to Prism. This will be the default Prism password. Click `Next`.
```
Is this supposed to be the new password I entered when I was forced to change the password?  Or the original default password from the Nutanix docs?

cprivitere replied:
~~It should be the default password.~~
~~I was wrong about this. It needs to be the new password you set on Prism.~~
I was wrong about both. Neither is working. This is weird.

cprivitere replied:
I've added a new workflow that has you update the passwrod on prism central and THEN register prism to it.

cprivitere committed:
fix: Errors in two steps
- Add mention of extra Deploy screen when deploying Prism Central
- Add new workflow to update Prism Central password before registering Prism with Prism Central.

Signed-off-by: Chris Privitere <23177737+cprivitere@users.noreply.github.com>

cprivitere committed:
chore: add numbers to parts 5 and 6 and update index

Signed-off-by: Chris Privitere <23177737+cprivitere@users.noreply.github.com>

cprivitere committed:
fix: error in step 7 referring to which password to use.

Signed-off-by: Chris Privitere <23177737+cprivitere@users.noreply.github.com>

cprivitere committed:
fix: typo

Signed-off-by: Chris Privitere <23177737+cprivitere@users.noreply.github.com>

cprivitere committed:
fix: errors in kubernetes workflow, updates to a couple images

Signed-off-by: Chris Privitere <23177737+cprivitere@users.noreply.github.com>

cprivitere committed:
fix: use separte window for prism central

Signed-off-by: Chris Privitere <23177737+cprivitere@users.noreply.github.com>

ctreatma commented on mkdocs.yml:
```diff
@@ -15,8 +15,9 @@ nav:
-  - Conclusion: parts/conclusion.md
+      - 5. Prism Central: parts/5-prism_central.md
+      - 6. Kubernetes: parts/6-kubernetes.md
+      - Conclusion: parts/conclusion.md
```
The `Conclusion` should stay at its old indentation; it's intended to show up outside of the `Parts` dropdown in the left nav.

ctreatma commented on docs/parts/6-kubernetes.md:
```diff
@@ -0,0 +1,105 @@
+
+### 1. Open Prism UI
+
+Let's login to the Prism Central UI from your computer at `https://localhost:9440`. Note that the Prism UI ships with a self-signed TLS certificate, so most browsers will display a security warning. This security warning is unavoidable. If your browser will allow you to ignore the warning and load the site, do that; if not, try a different browser.
```
```suggestion
Let's login to the Prism Central UI from your computer at `https://localhost:19440`. Note that the Prism UI ships with a self-signed TLS certificate, so most browsers will display a security warning. This security warning is unavoidable. If your browser will allow you to ignore the warning and load the site, do that; if not, try a different browser.
```

Should this be 19440?

ctreatma commented on docs/parts/5-prism_central.md:
```diff
@@ -0,0 +1,100 @@
+
+### 11. Wait for update to complete and continue to next Part
+
+In the next part we'll deploy another piece of the Nutanix stack onto the cluster.
```
I didn't have to wait for anything after putting in the name servers and the NTP server.  Did I miss a step to upgrade something?

cprivitere replied:
Nope, I removed updating from this section and forgot to delete this line.

ctreatma commented on docs/parts/6-kubernetes.md:
```diff
@@ -0,0 +1,105 @@
+
+### 3. Enable Kubernetes Engine
+
+Click on the three lined menu in the upper left of the Prism Central UI and select `Services` -> `Kubernetes`. Then click `Enable Karbn` or `Enable Kubernetes`. Which one you'll see depends on the version of the Nutanix Kubernetes Engine you have installed, which we'll be updating anyway in the next step. After a while it will start up and tell you there are no kubernetes clusters. That's fine, c ...
```
```suggestion
Click on the three lined menu in the upper left of the Prism Central UI and select `Services` -> `Kubernetes`. Then click `Enable Karbon` or `Enable Kubernetes`. Which one you'll see depends on the version of the Nutanix Kubernetes Engine you have installed, which we'll be updating anyway in the next step. After a while it will start up and tell you there are no kubernetes clusters. That's fine, continue on to the next step.
```

ctreatma commented on docs/parts/6-kubernetes.md:
```diff
@@ -0,0 +1,105 @@
+
+### 2. Update the Cluster
+
+Navigate to LCM take an Inventory and wait for it to finish.
```
I think there are some details missing here.  I'm assuming I should find and click the `Perform Inventory` button; if that's the correct button, can/should I ignore the auto inventory checkbox?

cprivitere committed:
fix: remove old reference to updating, remove discussion questions about updating

Signed-off-by: Chris Privitere <23177737+cprivitere@users.noreply.github.com>

cprivitere committed:
fix: Expand LCM details and fix typoes and port references.

Signed-off-by: Chris Privitere <23177737+cprivitere@users.noreply.github.com>

ctreatma commented on docs/parts/6-kubernetes.md:
```diff
@@ -0,0 +1,105 @@
+
+![K8s Demo Network Provider](../images/k8s-demo-network-provider.png)
+
+On the Storage Class page, fill in the cluster username and password, leave the rest of the options as their defaults, and click `Create`.
```
I wasn't prompted for a username and password here

cprivitere committed:
docs: Separate Conclusion on the navigation dropdown

Signed-off-by: Chris Privitere <23177737+cprivitere@users.noreply.github.com>

cprivitere committed:
fix: storage class doesn't mention username/password on current version of NKE

Signed-off-by: Chris Privitere <23177737+cprivitere@users.noreply.github.com>

ctreatma commented on docs/parts/6-kubernetes.md:
````diff
@@ -0,0 +1,105 @@
+Transfer it to your bastion host:
+
+```sh
+scp k8s-demo-kubectl.cfg -i $(terraform output -raw ssh_private_key) root@$(terraform output -raw bastion_public_ip):~/kubeconfig
````
```suggestion
scp -i $(terraform output -raw ssh_private_key) <path-to-downloads>/k8s-demo-kubectl.cfg root@$(terraform output -raw bastion_public_ip):~/kubeconfig
```

In my case `k8s-demo-kubectl.cfg` wasn't downloaded to the working directory; not sure how best to capture that potential variation.

ctreatma commented on docs/parts/6-kubernetes.md:
````diff
@@ -0,0 +1,105 @@
+Then, set the `KUBECONFIG` environment variable to point to the kubeconfig file:
+
+```sh
+export KUBECONFIG=$(PWD)/kubeconfig
````
```suggestion
export KUBECONFIG=$(pwd)/kubeconfig
```

ctreatma commented on docs/parts/6-kubernetes.md:
````diff
@@ -0,0 +1,105 @@
+export KUBECONFIG=$(PWD)/kubeconfig
+```
+
+Now you can use `kubectl` to interact with your Kubernetes cluster:
````
I had to `snap install kubectl --classic`, but maybe I ran an older version of the TF module?

cprivitere committed:
Update docs/parts/6-kubernetes.md

Co-authored-by: Charles Treatman <ctreatman@equinix.com>

cprivitere committed:
Update docs/parts/6-kubernetes.md

Co-authored-by: Charles Treatman <ctreatman@equinix.com>

cprivitere committed:
docs: add directions to install kubectl

Signed-off-by: Chris Privitere <23177737+cprivitere@users.noreply.github.com>

ctreatma reviewed (approved):
![It's done](https://media4.giphy.com/media/v1.Y2lkPTc5MGI3NjExNmE2aDE4dXd1b2NqMjcyN2l6NHoxdzhobjh0eXhtano5Y2E2dzBxNiZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/3oKIPf3C7HqqYBVcCk/giphy.gif)
